### PR TITLE
Add new item to shopping list

### LINKED ISSUE #4: 4. As a user, I want to add a new item to my shopping list so I can start recording purchases
## Summary
Users need to be able to add items to their shopping lists. To do this, the `AddItem` view should present them with a form that allows them to enter two things:
1. The name of the item, and
2. An estimate of when they think they’ll need to buy the item again.

[The README file for the shopping list API](https://github.com/the-collab-lab/smart-shopping-list-next/blob/main/src/api/README.md) will be helpful as you work on this feature.

## Acceptance criteria
UI-related tasks:
- [x] The `Home` view displays a form that allows them to enter the name of the item and select how soon they anticipate needing to buy it again. There should be 3 choices for this:
	- “Soon”, corresponding to 7 days
	- “Kind of soon”, corresponding to 14 days
	- “Not soon”, corresponding to 30 days
- [x] The input that accepts the name of the item has a semantic `label` element associated with it
- [x] The user can submit this form with both the mouse and the `Enter` key  
- [x] When the user submits the form, they see a message indicating that the item either was or was not saved to the database.

Data-related tasks:
- [x] The `console.log` in the `addItem` function in `src/api/firebase.js`  is replaced with a function that adds the new document to the Firestore database. That function will be imported from the `firebase/firestore` module.
- [x]  The user’s soon/not soon/kind of soon choice is used to calculate `nextPurchasedDate`

## Notes
- When this feature is implemented correctly, new items will *automatically* show up in the list view because `streamListItem` gets new data from Firestore every time there’s a change.
- You will know your `<label>` is correctly implemented if you click on the label and keyboard focus moves to the related input. Refer to [the MDN docs on `<label>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label) for more info.
- Users will be able to submit the form with the `Enter` key if you listen for the right JavaScript event on the right element! If you're adding an `onKeyDown` listener anywhere, you might be overcomplicating things.
- The add item form has the potential to be a lot of scope. [The wireframe for the "add item" page](https://github.com/the-collab-lab/smart-shopping-list/blob/main/_resources/wireframes/add-item.png) has several notes related to accessibility and UX.

### PR DESCRIPTION
**Pull request by Terieyenike**
## Description
This PR enables users to add items to their shopping lists by connecting the client side to the firestore. The data is stored and rendered to the `List page` with a successful message to signify the process was implemented correctly. The item from each entered input saves to the items collection of the database.

## Related Issue

Closes #4 

## Acceptance Criteria
UI-related tasks:
- [x] The `Home` view displays a form that allows them to enter the name of the item and select how soon they anticipate needing to buy it again. There should be 3 choices for this:
	- “Soon”, corresponding to 7 days
	- “Kind of soon”, corresponding to 14 days
	- “Not soon”, corresponding to 30 days
- [x] The input that accepts the name of the item has a semantic `label` element associated with it
- [x] The user can submit this form with both the mouse and the `Enter` key  
- [x] When the user submits the form, they see a message indicating that the item was or was not saved to the database.

Data-related tasks:
- [x] The `console.log` in the `addItem` function in `src/api/firebase.js`  is replaced with a function that adds the new document to the Firestore database. That function will be imported from the `firebase/firestore` module.
- [x]  The user’s soon/not soon/kind of soon choice is used to calculate `nextPurchasedDate`

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
| ✓   | :sparkles: New feature     |
|  ✓  | :hammer: Refactoring       |

### Before
new list item

### After
updated new list add item page

## Testing Steps / QA Criteria

- From your terminal, pull down this branch with `git pull origin ve-te-add-newItem-to-shopping-list` and check that branch out with `git branch ve-te-add-newItem-to-shopping-list`.
- Then start the development server with `npm start` to launch the app.
- Fill out the input field with a new item. Ensure it displays a successful message that the item is saved to the database and a button to show the `list` page.

### CODE REVIEW

**Comment by Terieyenike** [comment]
@faysvas @PatrickNiyogitare28 @pmbanugo 
Please review my PR.

Thank you!

**Comment by Terieyenike** [comment]
@Mama-simba @thatgirldorian please check our work.
Thank you!

**Comment by Mama-simba** [comment]
@Terieyenike and @mohanner-coder good job! Functionality-wise is good. Items get added randomly on any place of the list by mouse click or keyboard.
I would suggest to fill in your PR with some description and your acceptance criteria.

**Comment by faysvas** [comment]
@Terieyenike @mohanner-coder Please update the PR description section so we can understand what the PR is about :slightly_smiling_face:

File: src/views/AddItem.jsx [hunk 1]
```diff
@@ -1,3 +1,109 @@
...
+					<div className="select-next-purchase-buttons">
+						<fieldset>
+							<legend>How soon will you buy this again?</legend>
+							<div className="radio-btn">
+								<label htmlFor="soon">
+									<input
+										type="radio"
+										name="daysUntilNextPurchase"
+										id="soon"
+										value={7 || daysUntilNextPurchase}
```
Comment (faysvas) [inline src/views/AddItem.jsx:70]: Why is an OR needed here `7 || daysUntilNextPurchase`? The value is something that doesn't change but always remains the same for each radio button :slightly_smiling_face:

File: src/api/firebase.js [hunk 2]
```diff
@@ -50,7 +50,7 @@ export async function addItem(listId, { itemName, daysUntilNextPurchase }) {
 	const listCollectionRef = collection(db, listId);
 	// TODO: Replace this call to console.log with the appropriate
 	// Firebase function, so this information is sent to your database!
```
Comment (thepatrickniyo) [inline src/api/firebase.js]: You do not need the comments anymore

File: src/views/AddItem.jsx (hunk 1 above)
Comment (Terieyenike) [reply src/views/AddItem.jsx:70]: @faysvas, it is a fallback to the initial state `daysUntilNextPurchase`.

**Comment by Terieyenike** [comment]
Hi @faysvas @PatrickNiyogitare28,
I have updated the PR description and its other details.

Thank you

File: src/api/firebase.js (hunk 2 above)
Comment (thepatrickniyo) [reply src/api/firebase.js]: @Terieyenike please remember to remove the TODO comments whenever you are done with implementing the requirement

**Review by thepatrickniyo — CHANGES_REQUESTED** [review]
Please address to my feedback. You need to remove the TODO comments for the best code quality. 

You need to remove the TODO comments since you have implemented what they requested

File: src/api/firebase.js (hunk 2 above)
Comment (Terieyenike) [reply src/api/firebase.js]: I forgot again. So sorry.

Comment (Terieyenike) [reply src/api/firebase.js]: @PatrickNiyogitare28 Done.

**Pull request merged**